pkg/dataobj/metastore/object.go

```diff
@@ -134,13 +165,16 @@ func (m *ObjectMetastore) Streams(ctx context.Context, start, end time.Time, mat
 	level.Debug(utillog.WithContext(ctx, m.logger)).Log("msg", "ObjectMetastore.Streams", "tenant", tenantID, "start", start, "end", end, "matchers", matchersToString(matchers))
 
 	// Get all metastore paths for the time range
-	var storePaths []string
-	for path := range iterStorePaths(tenantID, start, end) {
+	var (
+		storePaths []string
+		prefix     = storagePrefixFor(m.cfg, tenantID)
```

Comment:
Would it be simpler to have a function which returns the bucket for this operation?
For each call, you could extract the tenant ID and if indexes are enabled you could return a prefixed bucket, else the regular bucket. Effectively `chroot`ing the bucket filepath.

That way you shouldn't need to store prefixed paths in the top-level metastore or apply prefixes every time you access something, the bucket would automatically do it for you.